src/parser/shaman/restoration/modules/talents/AncestralProtectionTotem.tsx

```diff
@@ -13,26 +13,31 @@ import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
 
 import Analyzer from 'parser/core/Analyzer';
 import Combatants from 'parser/shared/modules/Combatants';
-import { EventType } from 'parser/core/Events';
+import { Event, EventType, HasAbility, HasSource } from 'parser/core/Events';
+import { WCLEventsResponse, WclOptions } from 'common/WCL_TYPES';
 
 //RQXjBJ1kG9pAC2DV/21-Mythic++Atal'Dazar+-+Kill+(51:52)/Koorshaman/
 
 class AncestralProtectionTotem extends Analyzer {
   static dependencies = {
     combatants: Combatants,
   };
+
+  protected combatants!: Combatants;
+
   loaded = false;
-  aptEvents = [];
-  constructor(...args) {
-    super(...args);
+  aptEvents: Array<Event<EventType.Cast | EventType.ApplyDebuff>> = [];
```

Comment:
Does `Array<CastEvent | ApplyDebuffEvent>` work here or does it scream at you?

Reply:
Does work, which also means that I can remove one of the ifs further down. 👍